src/bots/marketBidAskTwapCrank.ts

```diff
@@ -0,0 +1,244 @@
+	private intervalIds: Array<NodeJS.Timer> = [];
+	private userMap: UserMap;
+	private driftEnv: DriftEnv;
+
+	private dlob: DLOB;
+	private latestDlobSlot: number;
+	private lastDlobRefreshTime = 0;
+	private lookupTableAccount: AddressLookupTableAccount;
+	private pollingIntervalMs: number;
+
+	private watchdogTimerMutex = new Mutex();
+	private watchdogTimerLastPatTime = Date.now();
+
+	constructor(
+		driftClient: DriftClient,
+		slotSubscriber: SlotSubscriber,
+		driftEnv: DriftEnv,
+		config: BaseBotConfig
+	) {
+		this.slotSubscriber = slotSubscriber;
+		this.name = config.botId;
+		this.dryRun = config.dryRun;
+		this.runOnce = config.runOnce || false;
+		this.driftClient = driftClient;
+		this.driftEnv = driftEnv;
+	}
+
+	public async init() {
+		logger.info(`${this.name} initing`);
+		this.lookupTableAccount =
+			await this.driftClient.fetchMarketLookupTableAccount();
+
+		// initialize userMap instance
+		this.userMap = new UserMap(
+			this.driftClient,
+			this.driftClient.userAccountSubscriptionConfig,
+			false
+		);
+		await this.userMap.subscribe();
+
+		this.pollingIntervalMs = 1000;
+	}
+
+	public async reset() {
+		for (const intervalId of this.intervalIds) {
+			clearInterval(intervalId);
+		}
+		this.intervalIds = [];
+
+		await this.userMap.unsubscribe();
+	}
+
+	public async startIntervalLoop(intervalMs: number): Promise<void> {
+		logger.info(`${this.name} Bot started!`);
+		if (this.runOnce) {
+			await this.tryTwapCrank();
+		} else {
+			const intervalId = setInterval(this.tryTwapCrank.bind(this), intervalMs);
+			this.intervalIds.push(intervalId);
+		}
+	}
+
+	public async healthCheck(): Promise<boolean> {
+		let healthy = false;
+		await this.watchdogTimerMutex.runExclusive(async () => {
+			healthy =
+				this.watchdogTimerLastPatTime > Date.now() - 2 * this.defaultIntervalMs;
+		});
+		return healthy;
+	}
+
+	private async initDlob2() {
+		try {
+			this.latestDlobSlot = this.slotSubscriber.currentSlot;
+			this.dlob = await this.userMap.getDLOB(this.slotSubscriber.currentSlot);
+			this.lastDlobRefreshTime = Date.now();
+		} catch (e) {
+			logger.error(`Error loading dlob: ${e}`);
+			this.lastDlobRefreshTime = 0;
+		}
+	}
+
+	private getCombinedList(makersArray) {
+		const combinedList = [];
+
+		for (const maker of makersArray) {
+			const uA = this.userMap.getUserAuthority(maker.toString());
+			if (uA !== undefined) {
+				const uStats = getUserStatsAccountPublicKey(
+					this.driftClient.program.programId,
+					uA
+				);
+
+				// Combine maker and uStats into a list and add it to the combinedList
+				const combinedItem = [maker, uStats];
+				combinedList.push(combinedItem);
+			} else {
+				console.log(
+					'skipping maker... cannot find authority for userAccount=',
+					maker.toString()
+				);
+			}
+		}
+
+		return combinedList;
+	}
+
+	private async tryTwapCrank() {
+		await this.init();
+		await this.initDlob2();
+
+		const ixs = [];
+
+		const crankMarkets = [0, 1, 2, 5, 8];
```

Comment:
intentionally not running it for all mkts?

Reply:
yeah should make a couple bundles, was just testing

Reply:
Possible to make this crank all markets by default? Maybe user can pass an arg to for subset

Reply:
just made it loop over every perp market in chunks